In [10]:
from astropy.table import Table, vstack
from astroquery.vizier import Vizier
import astropy.units as u
import astropy.coordinates as coords
import pandas as pd
import numpy as np

In [5]:
# Convertir a Astropy Table para facilitar el uso con Astroquery
table = pd.read_csv('/home/seba/Documents/CG/Hernandez-Fernandez/Galaxy_catalogue.csv')
coords_table = Table.from_pandas(table)

In [19]:
# Define el catálogo
catalog = "II/328/allwise"

# Configura VizieR
v = Vizier(columns=['RAJ2000', 'DEJ2000', 'W1mag', 'W2mag', 'W3mag', 'W4mag'],
           row_limit=-1)  # Sin límite de filas, ajusta según necesidad

# Lista para almacenar resultados
results = []

# Bucle sobre cada fila en la tabla de coordenadas
for row in coords_table:
    c = coords.SkyCoord(ra=row['_RAJ2000']*u.degree, dec=row['_DEJ2000']*u.degree, frame='icrs')
    result = v.query_region(c, radius=2*u.arcsec, catalog=catalog)
    if result:
        if isinstance(result, list):
            result = result[0]
        results.append(result)

if results:
    combined_results = vstack(results)
    # Imprime la tabla combinada para revisarla
    print(combined_results)
    
    combined_results.write('wise_data.csv', format='csv')
else:
    print('No se obtuvieron resultados')

  RAJ2000     DEJ2000   W1mag  W2mag  W3mag  W4mag 
    deg         deg      mag    mag    mag    mag  
----------- ----------- ------ ------ ------ ------
  0.0793772 -22.5761554 15.130 14.868 11.450  8.595
  0.0912074 -22.5879573 16.224 16.048 12.339  8.382
  0.1004063 -22.5888837 14.308 14.103 10.370  8.101
  0.1115195 -22.5959599 15.558 15.137 11.501  8.356
  0.3129671  13.1131761 14.199 14.058 11.020  8.362
  0.3331462  13.1112154 10.273 10.167  6.922  4.483
  0.3726389  13.0985394 16.451 16.545 12.519  8.836
  2.8351978 -53.9557714 16.238 15.989 12.865  8.849
  2.8378998 -53.9811095 15.083 14.892 10.865  8.435
  2.8428158 -53.9539555 16.748 16.959 12.635  9.165
  2.8524137 -53.9310158 13.288 13.123  9.227  7.371
  3.3139385 -41.5710571 14.833 14.834 11.616  9.038
  3.3172848 -41.5955268 13.845 13.449  9.311  7.480
  3.3265429 -41.5732188 12.322 12.276 12.095  9.509
  3.3528167 -41.5784179 14.255 14.085 10.580  8.880
  3.5174332 -22.5528013 14.094 13.723 10.217  8.555
  3.5320742 

In [12]:
t = Table.read('Galaxies_IR_wise', format='ascii')

grouped_t = t.group_by('Group')

valid_groups=[]

for group in grouped_t.groups:
    if any (group['zmemb'] != 0):
        valid_groups.append(group['Group'][0])

filtered_table = t[np.isin(t['Group'], valid_groups)]


$log SFR_{NUV} (M_{\odot}yr^{-1}) = log L_{NUV}, corr(L_{\odot}) -9.33$


$log SFR_{FUV} (M_{\odot}yr^{-1}) = log L_{FUV}, corr(L_{\odot}) -9.51$